src/libraries/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.T.cs

```diff
@@ -205,6 +205,13 @@ public void GetMemory_DefaultCtor(int sizeHint)
             Assert.Equal(sizeHint <= 256 ? 256 : sizeHint, memory.Length);
         }
 
+        [Fact]
+        public void GetMemory_ExceedMaximumBufferSize()
+        {
+            var output = new ArrayBufferWriter<T>(256);
+            Assert.Throws<OutOfMemoryException>(() => output.GetMemory(int.MaxValue));
```

Comment:
Are all the new branches of `CheckAndResizeBuffer` covered by this/other unit tests? Can you please verify by doing a quick code coverage run, locally?

Reply:
Now it is :)